src/pages/add-products/components/product-description-form/index.tsx

```diff
@@ -0,0 +1,261 @@
+import { gendersOptions } from "./constants.ts";
+import { productDescriptionValidation } from "./validation.ts";
+
+const ProductDescriptionForm: React.FC = () => {
+	const { t } = useTranslation();
+	const navigate = useNavigate();
+	const dispatch = useAppDispatch();
+
+	const { brand, description, gender, material, name } = useAppSelector(
+		(state: RootState) => state.productForm.productDescription,
+	);
+	const { data: brands } = useGetProductBrandsQuery(undefined);
+	const { data: materials } = useGetProductMaterialsQuery(undefined);
+
+	const [productDescription, setProductDescription] = useState(description);
+	const [selectedBrand, setSelectedBrand] = useState<null | number>(brand);
+	const [selectedMaterial, setSelectedMaterial] = useState<null | number>(
+		material,
+	);
+	const [selectedGender, setSelectedGender] = useState<null | number>(gender);
+
+	const { control, errors, handleSubmit, setValue } =
+		useAppForm<ProductDescription>({
+			defaultValues: {
+				brand,
+				description,
+				gender,
+				material,
+				name,
+			},
+			validationSchema: productDescriptionValidation,
+		});
+
+	const handleInputChange = useCallback(
+		async (formData: ProductDescription): Promise<void> => {
+			dispatch(updateProductDescription(formData));
+			navigate(AppRoute.PRODUCT_VARIANTS);
+		},
+		[dispatch, navigate],
+	);
+
+	const handleBrandChange = useCallback(
+		(event: SelectChangeEvent<number>) => {
+			const brandId = +event.target.value;
+			setSelectedBrand(brandId);
+			setValue("brand", brandId);
+		},
+		[setValue],
+	);
+
+	const handleMaterialChange = useCallback(
+		(event: SelectChangeEvent<number>) => {
+			const materialId = +event.target.value;
+			setSelectedMaterial(materialId);
+			setValue("material", +event.target.value);
+		},
+		[setValue],
+	);
+
+	const handleGenderChange = useCallback(
+		(event: SelectChangeEvent<number>) => {
+			const gender = +event.target.value;
+			setSelectedGender(gender);
+			setValue("gender", gender);
+		},
+		[setValue],
+	);
+
+	const handleChangeName = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>) => {
+			const name = event.target.value;
+			setValue("name", name);
+		},
+		[setValue],
+	);
+
+	const handleChangeDescription = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>) => {
+			const description = event.target.value;
+			setProductDescription(description);
+			setValue("description", description);
+		},
+		[setValue],
+	);
+
+	const handleFormSubmit = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			event_.preventDefault();
+			void handleSubmit(handleInputChange)(event_);
+		},
+		[handleSubmit, handleInputChange],
+	);
+
+	const brandsOptions: OptionType[] = brands
+		? brands.map((brand) => ({
+				label: brand.name,
+				value: brand.id,
+		  }))
+		: [];
+
+	const materialsOptions: OptionType[] = materials
+		? materials.map((material) => ({
+				label: material.name,
+				value: material.id,
+		  }))
+		: [];
```

Comment:
const materialsOptions: OptionType[] = materials?.map((material) => ({
				label: material.name,
				value: material.id,
		  })) || [];

Reply:
I think it should work